Add default variable logic for particle renderer

sgpearse opened the pull request:
Fixes #3246 by adding new logic for selecting default field variables.  If Vapor cannot detect X and Y variables beginning with U or V, then we search for variables that end in X, Y, and (if 3d) Z.

This applies to the use cases of 1) magnetic fields (bx, by, and bz) and our DCP standard naming convention (vel_x, vel_y, and vel_z).

StasJ reviewed (commented):
This is a nice feature but I don't see how it fixes the bug? The particle renderer will still fail when the velocity variables are not set, even if they are not used.

sgpearse commented:
I see what you mean.  I added a condition to ParticleRenderer.cpp:320 that prevents reading vector variables when direction is not enabled.  I'm not sure why git is saying I rewrote 28 lines in that file.  The condition at line 320 is what corrects the issue.
